blockprod/src/interface/blockprod_interface_impl.rs

```diff
@@ -42,6 +42,8 @@ impl BlockProductionInterface for BlockProduction {
             None => crate::detail::TransactionsSource::Mempool,
         };
 
-        self.generate_block(reward_destination, transactions).await
+        let (block, end_receiver) = self.produce_block(reward_destination, transactions).await?;
+        let _finished = end_receiver.await;
```

Comment:
Maybe this should be `end_receiver.await?;` just in case the other end closes/errors out

Reply:
The only error that can happen is if the channel is closed. We don't care about that. I'll add a comment about it.